operator/webhooks/pod_mutating_webhook.go

```diff
@@ -62,9 +64,10 @@ func (a *PodMutatingWebhook) InjectDecoder(d *admission.Decoder) error {
 
 func (a *PodMutatingWebhook) isKeptnAnnotated(pod *corev1.Pod) bool {
 	_, gotApplicationAnnotation := pod.Annotations["keptn.sh/application"]
```

Comment:
We need some additional logic :-) If the application is empty, use the workload (service) name. If the version is empty, calculate a hash of the pod spec (at least container/image, environment) and use this as the version. Add these annotations to the pod (therefore the scheduler can do its job)

Reply:
Therefore, we could return true if only one keptn annotation exists.

Reply:
added